tensorrt_llm/_torch/auto_deploy/custom_ops/torch_attention.py

```diff
@@ -228,6 +228,33 @@ def apply_rotary_pos_emb_ds(q, k, cos, sin, position_ids, unsqueeze_dim=1):
     return q_embed, k_embed
 
 
+@torch.library.custom_op("rope::apply_rope_ds", mutates_args=())
+def apply_rope_ds(
+    q: torch.Tensor,
+    k: torch.Tensor,
+    cos: torch.Tensor,  # [B, 1, seq_len, head_dim]
+    sin: torch.Tensor,  # [B, 1, seq_len, head_dim]
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    """
+    DS-style RoPE: interleaves Q/K channels, indexes cos/sin by position_ids
+    and returns rotated (q_embed, k_embed).
```

Comment:
There is no cos/sin indexing here with position_ids since it's done outside the op. Please update the comment accordingly